probe-rs/src/architecture/arm/core/exception_handling/armv7m.rs

```diff
@@ -0,0 +1,309 @@
+
+memory_mapped_bitfield_register! {
+    /// HFSR - HardFault Status Register
+    pub struct Hfsr(u32);
+    0xE000ED2C, "HFSR",
+    impl From;
+    debug_event, _: 31;
+    escalation_forced, _: 30;
+    vector_table_read_fault, _: 1;
+}
+
+memory_mapped_bitfield_register! {
+    /// CFSR - Configuratble Status Register (UFSR[31:16], BFSR[15:8], MMFSR[7:0])
+    pub struct Cfsr(u32);
+    0xE000ED28, "CFSR",
+    impl From;
+    /// When SDIV or UDIV instruction is used with a divisor of 0, this fault occurs if DIV_0_TRP is enabled in the CCR.
+    uf_div_by_zero, _: 25;
+    /// Multi-word accesses always fault if not word aligned. Software can configure unaligned word and halfword accesses to fault, by enabling UNALIGN_TRP in the CCR.
+    uf_unaligned_access, _: 24;
+    /// A coprocessor access error has occurred. This shows that the coprocessor is disabled or not present.
+    uf_coprocessor, _: 19;
+    /// An integrity check error has occurred on EXC_RETURN.
+    uf_integrity_check, _: 18;
+    /// Instruction executed with invalid EPSR.T or EPSR.IT field.
+    uf_invalid_state, _: 17;
+    /// The processor has attempted to execute an undefined instruction. This might be an undefined instruction associated with an enabled coprocessor.
+    uf_undefined_instruction, _: 16;
+    /// BFAR has valid contents.
+    bf_address_register_valid, _: 15;
+    /// A bus fault occurred during FP lazy state preservation.
+    bf_fp_lazy_state_preservation, _: 13;
+    /// A derived bus fault has occurred on exception entry.
+    bf_exception_entry, _: 12;
+    /// A derived bus fault has occurred on exception return.
+    bf_exception_return, _: 11;
+    /// Imprecise data access error has occurred.
+    bf_imprecise_data_access_error, _: 10;
+    /// Precise data access error has occurred.
+    bf_precise_data_access_error, _: 9;
+    ///  A bus fault on an instruction prefetch has occurred. The fault is signalled only if the instruction is issued.
+    bf_instruction_prefetch, _: 8;
+    ///  MMAR has valid contents.
+    mm_address_register_valid, _: 7;
+    /// A MemManage fault occurred during FP lazy state preservation.
+    mm_fp_lazy_state_preservation, _: 5;
+    /// A derived MemManage fault occurred on exception entry.
+    mm_exception_entry, _: 4;
+    /// A derived MemManage fault occurred on exception return.
+    mm_exception_return, _: 3;
+    ///  Data access violation. The MMAR shows the data address that the load or store tried to access.
+    mm_data_access_violation, _: 1;
+    ///  MPU or Execute Never (XN) default memory map access violation on an instruction fetch has occurred.
+    mm_instruction_fetch_violation, _: 0;
+}
+
+impl Cfsr {
+    /// Additional information about a Usage Fault, or Ok(None) if the fault was not a Usage Fault.
+    fn get_uf_details<T: CoreInterface>(&self, _core: &mut T) -> Result<Option<String>, Error> {
+        let source = if self.uf_coprocessor() {
```

Comment:
I would rather we had a proper enum here instead of returning strings.

The enum can then implement `ToString`.

Reply:
Tnhe `ToString()` will require all the same code of converting the enum into proper messsages. Also, have aook at the similar functions for `get_bf_details() and `get_mmf_details()`  ... th e enum `ToString()` will get even weirder when there are additional details like address locations to be appended to the strings.

It feels to me like it will add code, but not value, because it is not used anywhere other than in this module.

Reply:
Ok, I am okay keeping it like this.

Reply:
Cool. I updated the function names to include the word "description", to make it obvious that it returns text. 

Currently there is no code that requires more than a text description of this.